python/paddle/v2/fluid/optimizer.py

```diff
@@ -33,16 +33,44 @@ class Optimizer(object):
     but need to use one of it's implementation.
     """
 
-    def __init__(self, global_step=None, regularization=None):
+    def __init__(self, learning_rate, global_step=None, regularization=None):
+        assert learning_rate is not None
         self._global_step = global_step
         self.regularization = regularization
+        self._learning_rate = learning_rate
+        self._global_lr_var = None
         # Dictionary of accumulators. Some optimizer subclasses need to
         # allocate and manage extra variables associated with the parameters
         # to train. These variables are called accumulators.
         # {accum_name : { paramter_name : accumulator_for_parameter, ...}, ...}
         self._accumulators = defaultdict(lambda: dict())
         self.helper = None
 
+    def _create_global_learning_rate(self):
+        if isinstance(self._learning_rate, float):
+            self._global_lr_var = self.helper.create_global_variable(
+                name=unique_name("learning_rate"),
+                dtype='float32',
+                shape=[1],
+                lod_level=1,
+                persistable=True)
+            self.helper.set_variable_initializer(
+                var=self._global_lr_var,
+                initializer=Constant(self._learning_rate))
+        elif isinstance(self._learning_rate, framework.Variable):
+            self._global_lr_var = self._learning_rate
+        else:
+            raise ValueError("learning rate should be a float or a Variable, "
+                             "actual type is %s", type(self._global_lr_var))
+
+    @property
+    def learning_rate(self):
```

Comment:
Maybe `global_learning_rate()` is more accuracy.

Reply:
done